src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,520 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + RC_KINDS_TYPES
+module stdlib_stats_distribution_uniform
+    use stdlib_kinds
+    use stdlib_error, only : error_stop
+    use stdlib_stats_distribution_PRNG, only : dist_rand
+
+    implicit none
+    private
+
+    real(dp), parameter  :: MESENNE_NUMBER = 1.0_dp / (2.0_dp ** 53 - 1.0_dp)
+    integer(int64), parameter :: INT_ONE = 1_int64
+
+    public :: uniform_distribution_rvs
+    public :: uniform_distribution_pdf
+    public :: uniform_distribution_cdf
+    public :: shuffle
+
+
+    interface uniform_distribution_rvs
+    !! Version experimental
+    !!
+    !! Get uniformly distributed random variate for integer, real and complex
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
```

Comment:
for some reasons, these links do not seem to work properly.

Reply:
I followed the same format as PRNG module, can't find where the problem is, sorry not good at ford.

Reply:
No problems. I will check that once more, before merging